src/test/regress/sql/function_propagation.sql

```diff
@@ -756,6 +756,63 @@ SELECT create_distributed_function('func_to_colocate(int)');
 -- show that the pg_dist_object fields are gone
 SELECT distribution_argument_index, colocationid, force_delegation FROM pg_catalog.pg_dist_object WHERE objid = 'func_to_colocate'::regproc;
 
+
+-- Show that causing circular dependency via functions and default values are not allowed
+CREATE TABLE table_1_for_circ_dep(id int);
+select create_distributed_table('table_1_for_circ_dep','id');
+
+CREATE OR REPLACE FUNCTION func_1_for_circ_dep(col_1 table_1_for_circ_dep)
+RETURNS int
+LANGUAGE plpgsql AS
+$$
+BEGIN
+return 1;
+END;
+$$;
+
+CREATE TABLE table_2_for_circ_dep(id int, col_2 int default func_1_for_circ_dep(NULL::table_1_for_circ_dep));
+select create_distributed_table('table_2_for_circ_dep','id');
+CREATE OR REPLACE FUNCTION func_2_for_circ_dep(col_3 table_2_for_circ_dep)
+RETURNS int
+LANGUAGE plpgsql AS
+$$
+BEGIN
+return 1;
+END;
+$$;
+
+-- It should error out due to circular dependency
+ALTER TABLE table_1_for_circ_dep ADD COLUMN col_2 int default func_2_for_circ_dep(NULL::table_2_for_circ_dep);
+
+
+-- Show that causing circular dependency via functions and constraints are not allowed
+CREATE TABLE table_1_for_circ_dep_2(id int, col_1 int);
+select create_distributed_table('table_1_for_circ_dep_2','id');
+
+CREATE OR REPLACE FUNCTION func_1_for_circ_dep_2(param_1 int, param_2 table_1_for_circ_dep_2)
+RETURNS boolean
+LANGUAGE plpgsql AS
+$$
+BEGIN
+    return param_1 > 5;
+END;
+$$;
+
+CREATE TABLE table_2_for_circ_dep_2(id int, col_1 int check (func_1_for_circ_dep_2(col_1, NULL::table_1_for_circ_dep_2)));
+
+select create_distributed_table('table_2_for_circ_dep_2','id');
+CREATE OR REPLACE FUNCTION func_2_for_circ_dep_2(param_1 int, param_2 table_2_for_circ_dep_2)
+RETURNS boolean
+LANGUAGE plpgsql AS
+$$
+BEGIN
+    return param_1 > 5;
+END;
+$$;
+
+-- It should error out due to circular dependency
+ALTER TABLE table_1_for_circ_dep_2 ADD CONSTRAINT check_const_for_circ check (func_2_for_circ_dep_2(col_1, NULL::table_2_for_circ_dep_2));
```

Comment:
```SQL
create type t1 as (a int);
create type t2 as (a int, b t1);
ALTER TYPE t1 ADD ATTRIBUTE b t2;
ERROR:  composite type t1 cannot be made a member of itself

```
Maybe add this as well with a comment,  just in case PG supports this one day, we don't forget